### PR TITLE
SoM: Avoid destaking all nodes when there are no quality block producers

### PR DESCRIPTION
**Pull request by mvines**
The `--max-poor-block-producer-percentage` safely belt doesn't work when ALL nodes are classified as poor.   

Thankfully due to an RPC error, the https://buildkite.com/solana-labs/stake-o-matic/builds/456#86b06df8-014d-4330-9623-9817cfe8da56 job from yesterday failed instead of accidentally removing stake from all nodes

### CODE REVIEW

File: stake-o-matic/src/main.rs [hunk 1]
```diff
@@ -1134,8 +1134,11 @@ fn main() -> Result<(), Box<dyn error::Error>> {
     let (quality_block_producers, poor_block_producers, cluster_average_skip_rate) =
         classify_block_producers(&rpc_client, &config, last_epoch)?;
 
-    let too_many_poor_block_producers = poor_block_producers.len()
-        > quality_block_producers.len() * config.max_poor_block_producer_percentage / 100;
+    let too_many_poor_block_producers = if quality_block_producers.is_empty() {
+        true
+    } else {
+        poor_block_producers.len() > quality_block_producers.len() * config.max_poor_block_producer_percentage / 100
```
Comment (mvines) [inline stake-o-matic/src/main.rs:1140]: When quality_block_producers.is_empty(), we previously had:
`poor_block_producers.len() > 0 * config.max_poor_block_producer_percentage / 100`

So no quality block producers means that `too_many_poor_block_producers` is false and we destake the world!

File: stake-o-matic/src/main.rs [hunk 2]
```diff
@@ -1134,8 +1134,12 @@ fn main() -> Result<(), Box<dyn error::Error>> {
     let (quality_block_producers, poor_block_producers, cluster_average_skip_rate) =
         classify_block_producers(&rpc_client, &config, last_epoch)?;
 
-    let too_many_poor_block_producers = poor_block_producers.len()
-        > quality_block_producers.len() * config.max_poor_block_producer_percentage / 100;
+    let too_many_poor_block_producers = if quality_block_producers.is_empty() {
+        true
+    } else {
+        poor_block_producers.len()
+            > quality_block_producers.len() * config.max_poor_block_producer_percentage / 100
```
Comment (sakridge) [inline stake-o-matic/src/main.rs:1141]: extract to function & unit test?

**Review by sakridge — APPROVED** [review]
logic looks right

**Comment by sakridge** [comment]
Would it have been just bonus stake? or all stake?

File: stake-o-matic/src/main.rs (hunk 2 above)
Comment (mvines) [reply stake-o-matic/src/main.rs]: done

**Comment by mvines** [comment]
> Would it have been just bonus stake? or all stake?

just bonus I think, ~◎85m 😬

**Pull request merged**